### ISSUE TITLE
tuya-cli list-app waiting for request

### ISSUE THREAD

**Issue by jd0394** [AMoo-Miki/homebridge-tuya-lan#63]
I had to change my wifi network password and after updating the password on all my devices I had to remove and re add my 3 bulbs to get them back on my network and working on the tuya app. I then had my commands on the home app having no effect. I figured the issue is that I need to find the key for each bulb and add them to my config.json file for homebridge again. 

Ive done each step many times now but each time I'm stuck on "waiting for request" while refreshing the tuya app list of devices on my phone. I've made sure each step was done correctly on my phone, the profile is installed and certificate is verified in the about section of settings, and the manual proxy is properly enabled on my wifi settings. 

Any ideas what else I might be missing? I should also mention I'm very new to this. I have homebridge running on a MacBook and the tuya-cli list-app says proxy IP 192.168.0.24 port 8001, which is what I have entered into my phone.

Any help would be greatly appreciated, thanks.

**Comment by Dag0n** [comment]
Having the same issue here too.

**Comment by satelshawn** [comment]
Same issue here.

**Comment by carsten-h** [comment]
Same here!

**Comment by Rantaholic** [comment]
Sam we here. I wonder if they Tuya Smart app was updated.

**Comment by Rantaholic** [comment]
OK. I think they did update the app. I have successfully got the keys by using the CE Smart app (costco tuya plugs) instead which hasn’t been updated in a month. The Tuya and SmartLife apps were just updated and I couldn’t get them to work. Download now before they update it.

**Comment by carsten-h** [comment]
I have downloaded the CE Smart App and a few other older ones and tried them together with anyproxy on my Pi, but all of them are not working. :-(

**Comment by Rantaholic** [comment]
@carsten-h
I also updated to the latest version of the Tuya API:
npm i @tuyapi/cli -g
and the newest version of this module manually using:
npm i -g AMoo-Miki/homebridge-tuya-lan

Not sure if that made a difference. I’m on iOS as well so the apps might not be updated at the same time as android.

**Comment by carsten-h** [comment]
I installed tuyapi/cli two days ago, so it should be the newest. And I also installed the newest tuya-lan.

**Comment by PierreBrisorgueil** [comment]
Yep, same problem as you, switched from raspberry to macos,and Tuya to CE Smart and it's ok :) 

thx @Rantaholic

**Comment by KYDronePilot** [comment]
CE Smart has been updated and no longer works. From what I have seen, any of the Tuya apps listed as last being updated 2 months ago (appx. mid May, 2019) will no longer work.

I found another app that does still work, `Brilliant Smart`. Just keep in mind, this will not work in iOS 13 (at least in the betas).

**Comment by Abi-87** [comment]
I couldn't get any keys, I'm sure I did all steps right.
Any ideas?

**Comment by KYDronePilot** [comment]
@Abi-87 What app did you use to setup the devices?

**Comment by Abi-87** [comment]
I used many different apps like Tuyasmart, CE Smart, Brilliant Smart, but I didn't get any result. even I reinstalled home bridge twice and also I changed the Port and IP.

**Comment by KYDronePilot** [comment]
@Abi-87 Sorry, not sure what the issue could be. I had no trouble when following the instructions and using the Brilliant Smart app.

**Comment by adamgoose** [comment]
I ran into this as well. I was able to implement what the proxy and CLI are attempting to do simply by inspecting the traffic myself. 

1. Install the free version of [ProxyMan](https://proxyman.io/).
2. Open ProxyMan Settings, generate a new CA (automatic method works great). Optionally disable "Auto override the Proxy config in Network Preferences". This will make it easier to find the correct web request later.
3. Open Keychain Access, select the System keychain, find the "Proxyman CA", and export it to a file in .pem format.
4. Open this file on your iPhone somehow.
    - Use a file sharing app
    - Use iCloud Drive
    - Email it to yourself
5. Install the certificate, then open Settings -> General -> About -> Certificate Trust Settings (very bottom) and give it Full Trust.
6. Start ProxyMan, and note its "Listening Port" (9090 by default). Also note your local IP
7. Open Settings -> WiFi -> Blue "i" next to your connection -> Configure Proxy. Set your local IP and ProxyMan port.

At this point, your iPhone should be able to freely access the internet and all traffic will begin to appear in the ProxyMan UI. Just refresh your device list in the app a few times. 

Look for requests to `ay.tuyaus.com` (or something similar, based on your location I suppose). You might have to enable SSL interception on a request in order to see the payload. Refresh your device list again to see a raw payload!

![image](https://user-images.githubusercontent.com/611068/62268145-75b99700-b3f4-11e9-9969-a126c04612cf.png)

Search the raw response body for `tuya.m.my.group.device.list`. Use each devices `devId` and `localKey` for `id` and `key` respectively.

**Comment by verisgit** [comment]
Was not able to get the normal solution working to get the device IDs and Keys. Here's a solution for Windows 10 using Burp Suite (free community edition to capture the SSL device request) http://www.htgsd.com/information-technology/apple/homekit/how-to-capture-tuya-lan-homebridge-device-devid-and-key-on-windows-10/

**Comment by duguzuyang** [comment]
CE Smart app worked for me while Smart Life didn't

**Comment by verisgit** [comment]
Also I had to specify the version in order to get it working in my config.json

"version": "3.11"

**Comment by carsten-h** [comment]
Which version is this and where did you place it? Thank you!

**Comment by verisgit** [comment]
It’s the version reported in the Tuya app 
Me>Settings>About>Current aversions

**Comment by verisgit** [comment]
Note. I ignored the last number. Mine was 3.11.7

**Comment by carsten-h** [comment]
Thank you!
At which point of the config.json did you put it?

**Comment by verisgit** [comment]
You can see an example in my posts here https://github.com/AMoo-Miki/homebridge-tuya-lan/issues/7

**Comment by mkormendy** [comment]
The suggestion by @Rantaholic with the CE Smart app on iPhone Xs (latest iOS 12) worked for me!

**Comment by AMoo-Miki** [comment]
I have published a new method of obtaining the `id` and `key`. Please update to v1.4.0 and follow the [Setup Instructions](https://github.com/AMoo-Miki/homebridge-tuya-lan/wiki/Setup-Instructions).
